pkg/handlers/v1/attribute.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/assetattributor"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/domain"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/logs"
+)
+
+// AttributeHandler handles the Attribution endpoint for nexpose-asset-attributor
+type AttributeHandler struct {
+	AssetAttributor domain.AssetAttributor
+	LogFn           domain.LogFn
+	StatFn          domain.StatFn
+}
+
+// Handle processes the incoming domain.NexposeAssetVulnerabilities instance, queries available asset inventory systems via its
+// AssetAttributor, and returns a domain.AttributedAssetVulnerabilities instance, annotated with the business context of the asset
+// at Nexpose scan time.
+func (h *AttributeHandler) Handle(ctx context.Context, assetVulns domain.NexposeAssetVulnerabilities) (domain.NexposeAttributedAssetVulnerabilities, error) {
+	logger := h.LogFn(ctx)
+
+	attributedAssetVulns, err := h.AssetAttributor.Attribute(ctx, assetVulns)
+	if err != nil {
+		switch err.(type) {
+		case *assetattributor.AssetNotFoundError:
+			logger.Error(logs.AssetNotFoundError{Reason: err.Error()})
```

Comment:
> returns either a 404 Not Found response, or a 200 OK response with no results
based on the comment from this error, should this be an info level log? Also, should you have a template string in your api.yaml to return a 404? [example](https://github.com/asecurityteam/asset-inventory-api/blob/master/api.yaml#L141)